Yubico.Core/src/Yubico/PlatformInterop/Libraries.cs

```diff
@@ -12,14 +12,75 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+using System;
+
 namespace Yubico.PlatformInterop
 {
+    /// <summary>
+    /// Handles the loading and management of native libraries required by the Yubico SDK.
+    /// </summary>
+    /// <remarks>
+    /// This class provides cross-platform and cross-framework support for loading native libraries.
+    /// The implementation differs based on the target framework:
+    /// 
+    /// For .NET Framework 4.7:
+    /// - Native libraries must be placed in architecture-specific subdirectories (x86/x64)
+    /// - Library loading is handled explicitly at runtime based on process architecture
+    /// - Requires proper cleanup through the Cleanup method
+    /// 
+    /// For Modern .NET:
+    /// - Native libraries are handled by the runtime's built-in library loading mechanism
+    /// - Architecture-specific loading is managed automatically
+    /// - No explicit cleanup is required
+    /// </remarks>
     internal static partial class Libraries
     {
 #if NET47
+        /// <summary>
+        /// The filename of the native shims library for .NET Framework 4.7.
+        /// </summary>
+        /// <remarks>
+        /// For .NET Framework 4.7, the DLL must be placed in an architecture-specific subdirectory:
+        /// - x86/Yubico.NativeShims.dll for 32-bit processes
+        /// - x64/Yubico.NativeShims.dll for 64-bit processes
+        /// The correct version is loaded at runtime based on the process architecture.
+        /// </remarks>
         internal const string NativeShims = "Yubico.NativeShims.dll";
+
+        /// <summary>
+        /// Ensures the native library is properly loaded for .NET Framework 4.7.
+        /// </summary>
+        /// <exception cref="DllNotFoundException">
+        /// Thrown when the native library cannot be loaded. This could be due to:
+        /// - Missing DLL file in the architecture-specific directory (x86/x64)
+        /// - Incorrect architecture (x86/x64 mismatch)
+        /// - Missing dependencies
+        /// - Insufficient permissions
+        /// </exception>
+        /// <remarks>
+        /// This method must be called before any P/Invoke calls are made.
+        /// The implementation details are handled in Libraries.Net47.cs.
+        /// </remarks>
+        public static void EnsureInitialized() => Net47Implementation.Initialize();
 #else
```

Comment:
Since this is the same partial class, why not just keep all NET47 stuff in the NET47 class? You could rename this file to Libraries.NetStandard.cs or something to make the separation more obvious.

Reply:
Good catch